internal/imageservice/url_test.go

```diff
@@ -115,6 +126,43 @@ var _ = Describe("URL building", func() {
 	})
 })
 
+var _ = Describe("URL parsing", func() {
+	var (
+		baseURL = "https://image-service.example.com/v3"
+		version = "4.10"
+		arch    = "x86_64"
+		isoType = "full-iso"
+		id      = "2bf4b68e-c6cd-4603-9d79-1d05e5aa0665"
+	)
+
+	It("successfully parse /images/ URLs", func() {
+		u, err := ImageURL(baseURL, id, version, arch, isoType)
+		Expect(err).NotTo(HaveOccurred())
+
+		parsedISOType, parsedArch, parsedVersion, err := ParseDownloadURL(u)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(parsedISOType).To(Equal(isoType))
+		Expect(parsedArch).To(Equal(arch))
+		Expect(parsedVersion).To(Equal(version))
+	})
+
+	It("successfully parse shorter URLs", func() {
+		u, err := ShortImageURL(baseURL, ByAPIKeyPath, id, version, arch, isoType)
+		Expect(err).NotTo(HaveOccurred())
+
+		parsedISOType, parsedArch, parsedVersion, err := ParseDownloadURL(u)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(parsedISOType).To(Equal(isoType))
+		Expect(parsedArch).To(Equal(arch))
+		Expect(parsedVersion).To(Equal(version))
+	})
+
+	It("successfully parse malformed URL", func() {
```

Comment:
it should successfully parse a malformed URL?

Reply:
maybe I should call it `successfully parse a non-infraenv URL`? as the function is best-effort

Reply:
yeah, just making it clearer